Runtime/Tracking/Collision/CollisionNotifier.cs

```diff
@@ -191,9 +191,30 @@ public enum CollisionStates
         /// </summary>
         protected readonly EventData eventData = new EventData();
         /// <summary>
-        /// A reused instance to use when looking up <see cref="CollisionNotifier"/> components.
+        /// A reused instance to use when looking up <see cref="CollisionNotifier"/> components on start.
         /// </summary>
-        protected readonly List<CollisionNotifier> collisionNotifiers = new List<CollisionNotifier>();
+        protected readonly List<CollisionNotifier> startCollisionNotifiers = new List<CollisionNotifier>();
+        /// <summary>
+        /// A reused instance to use when looking up <see cref="CollisionNotifier"/> components on change.
+        /// </summary>
+        protected readonly List<CollisionNotifier> changeCollisionNotifiers = new List<CollisionNotifier>();
+        /// <summary>
+        /// A reused instance to use when looking up <see cref="CollisionNotifier"/> components on stop.
+        /// </summary>
+        protected readonly List<CollisionNotifier> stopCollisionNotifiers = new List<CollisionNotifier>();
+
+        /// <summary>
+        /// Whether the <see cref="collisionNotifiers"/> collection is being processed on start.
+        /// </summary>
+        protected bool processingStartNotifierCollection;

```

Comment:
```suggestion
        protected bool isProcessingStartNotifierCollection;
```
Same for the other two.